src/features/island/collectibles/components/Hourglass.tsx

```diff
@@ -177,6 +181,11 @@ export const Hourglass: React.FC<HourglassProps> = ({
   hourglass,
 }) => {
   const { gameService, showTimers } = useContext(Context);
+  const [
```

Comment:
It is a much better practice to implement `useSelector` here with a callback to get access just to the `state` value only. This follows the best practice recommendation of _least privilege - only accessing what you need_. Using `useActor` subscribes this component to ALL state changes in the machine which will then cause a re-render. We should always try to access only the data we need. I recommend refactoring all of these actors to selectors instead:

```ts
// Outside component
const _state = (state: MachineState) => state.context.state

// Inside component
const state = useSelector(gameService, _state)
```